frontend/src/io-managers/input.ts

```diff
@@ -131,7 +151,7 @@ export function createInputManager(editor: Editor, container: HTMLElement, dialo
 
 		if (!inTextInput) {
 			if (textInput) editor.instance.on_change_text(textInputCleanup(textInput.innerText));
-			else if (inCanvas) viewportPointerInteractionOngoing = true;
+			else viewportPointerInteractionOngoing = inCanvas instanceof Element;
```

Comment:
I think the "inCanvas" naming is a bit confusing at the moment because it can mean two different things without a clear distinction. First its declared as a boolean on the top level and then as a element inside some functions. Maybe the local variable names should be changes to "canvasElement"? The confusion was the mainly the result of scrolling through the diff and not seeing the  full code.

Reply:
Good point - I've renamed the top level one to `canvasFocused` and the other one to `isTargetingCanvas`

Reply:
Those names are much clearer, thanks both of you!